lib/hammer_cli_foreman/command_extensions/hosts/help/compute_resources.rb

```diff
@@ -0,0 +1,91 @@
+module HammerCLIForeman
+  module CommandExtensions
+    module Hosts
+      module Help
+        class ComputeResources < HammerCLI::CommandExtensions
+          class << self
+            attr_writer :current_providers, :add_host_specific_attrs,
+                        :attributes
+
+            def current_providers
+              @current_providers ||= HammerCLIForeman.compute_resources.keys
+              @current_providers
+            end
+
+            def add_host_specific_attrs?
+              @add_host_specific_attrs ||= true
+              @add_host_specific_attrs
+            end
+
+            def attributes
+              @attributes ||= :all
+              @attributes
+            end
+
+            def customized?
+              @customized
+            end
+
+            def customized=(boolean)
+              @customized ||= boolean
+            end
+
+            def custom(options = {})
+              new_self = Class.new(ComputeResources)
+              new_self.add_host_specific_attrs = options[:add_host_specific_attrs]
+              new_self.attributes = options[:attributes]
+              new_self.current_providers = options[:providers]
+              new_self.class_eval(help_block)
+              new_self.customized = true
+              new_self
+            end
+
+            def help_block
+              help_block = <<HELP_BLOCK
+                help do |h|
+                  h.section(_('Provider specific options')) do |h|
+                    HammerCLIForeman.compute_resources.each do |name, provider|
```

Comment:
I don't feel that putting this code under block is the right approach, I feel like it's not readable like that, what was the reason for changing this code?